docs/cli.rst

```diff
@@ -18,6 +18,38 @@ Below are a few sample command lines you might be interested in.
     If you have cloned Bowtie to :file:`/path/to/bowtie` you should be able to use ``$(ls /path/to/bowtie/implementations/ | sed 's/^| /-i /')`` in any command to expand out to all implementations.
     See `below <cli:running the official suite across all implementations>` for a full example.
 
+Add tab completion for implementations (and otherwise)
+------------------------------------------------------
+
+`Click Shell Completion Reference`_
+
+.. tabs::
```

Comment:
Like the click docs say, we should probably prefer suggesting that one saves the outputted contents here on their completion path rather than re-running it on each shell invocation, so maybe add a second line to each of these corresponding to the other box in the click docs.